sdk/internal/perf/perf_test.go

```diff
@@ -0,0 +1,80 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package perf
+
+import (
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func int64Ptr(i int64) *int64 {
+	return &i
+}
+
+type perfTest struct {
+	counter  int
+	setup    bool
+	teardown bool
+}
+
+func (p *perfTest) Setup()    { p.setup = true }
+func (p *perfTest) Run()      { p.counter++ }
+func (p *perfTest) TearDown() { p.teardown = true }
+
+func TestPerf(t *testing.T) {
```

Comment:
Are comments/documentation for each func not mandatory for Go?

Reply:
Not mandatory in the sense that the compiler will fail. But they should be in publicly exported functions. Go aims for 'self documenting code', but this test needs to be expanded anyways.